server/registry/internal/storage/list.go

```diff
@@ -515,20 +515,48 @@ func (c *Client) ListSpecRevisions(ctx context.Context, parent names.Spec, opts
 		return SpecList{}, status.Errorf(codes.InvalidArgument, "invalid page token %q: %s", opts.Token, err.Error())
 	}
 
+	if parent.ProjectID != "-" && parent.ApiID != "-" && parent.VersionID != "-" && parent.SpecID != "-" {
```

Comment:
Can we briefly explain what is happening here? (in a comment block above this line?) I know we've used this pattern before, but I find that the differences between != and == in this list of conditions are hard to visually parse, and without explanation this feels like a puzzle to work out (I think the answer is "complain about the lowest-level ancestor that doesn't exist")

Reply:
Done.